src/app-layer-ssl.c

```diff
@@ -2681,7 +2681,7 @@ static AppLayerResult SSLDecode(Flow *f, uint8_t direction, void *alstate,
     const uint8_t *init_input = input;
     int32_t input_len = (int32_t)StreamSliceGetDataLen(&stream_slice);
 
-    if (input == NULL &&
+    if ((input == NULL || input_len == 0) &&
```

Comment:
This looks ok to me. However, I have a few questions:
1. Can this not happen with other app protos?
2. Why is the data passed to a specific applayer parser if we know beforehand that the length of the data is 0?

Reply:
other protocols handle things in their own way, I've just check http and it just passes it on to libhtp

we send an empty message + set the EOF flags to tell the parser to close down

Reply:
ok. thank you very much! 🙇🏽‍♀️